### PR TITLE
Allowing unsetting of favourite program

### LINKED ISSUE #5132: 💻 Cannot unfavourite a program
After favouriting a program, there is no way to unfavourite it. It will just ask you the same question.

Screenshot 2024-02-12 at 10 25 51

### PR DESCRIPTION
**Pull request by TiBiBa**
Fixes #5132

**How to test:**
1. Login (have a public profile and a public program)
2. Click on favourite a program and unfavourite the program

https://github.com/hedyorg/hedy/assets/48122190/88bb0be2-16de-471b-ae10-92bfcaf07c55

### CODE REVIEW

**Comment by Felienne** [comment]
Hi @hasan-sh!

This is a somewhat longstanding R, could you pick it up? I am not sure how ready it is but I think it can be a small PR in between other things.

**Comment by hasan-sh** [comment]
> Hi @hasan-sh!
> 
> This is a somewhat longstanding R, could you pick it up? I am not sure how ready it is but I think it can be a small PR in between other things.

Will do so after the autosave PR. I'm also investigating the login problems that we discussed with the teachers!

**Review by hasan-sh — CHANGES_REQUESTED** [review]
I cannot favourite a program without a refresh. I'll review the code once this is done! Let me know if you need any help:)

**Comment by Annelein** [comment]
@hasan-sh all tests have passed, ready to be reviewed again, only needs to wait for that issue right?

**Comment by Annelein** [comment]
@hasan-sh I've also added a test for favouriting and unfavouriting a program as mentioned in #4251, can you let me know if this is how a test is supposed to be?

**Comment by hasan-sh** [comment]
Great @Annelein , will review your PR

File: templates/htmx-program.html [hunk 1]
```diff
@@ -26,26 +26,31 @@ <h3 id="submitted_header_{{ loop_index }}" class="mt-0 mb-4 text-red-500 text-ce
...
         </table>
         {% if public_profile and not second_teachers_programs %}
-            <div class="ml-36 favourite_program_container {% if public_profile.favourite_program and public_profile.favourite_program == program.id %}text-yellow-400{% else %}text-white{% endif %} text-2xl{% if not program.public %} hidden{% endif %}" id="favourite_program_container_{{ loop_index }}">
-                <i class="fa fa-star p-4 bg-blue-300 rounded-full cursor-pointer hover:bg-blue-600" onclick="hedyApp.set_favourite_program('{{ program.id }}', {{ loop_index }}, '{{_('favourite_confirm')}}');" aria-hidden="true"></i>
+            {% set favourite_program = public_profile.favourite_program == program.id %}
+            <div
+                class="ml-auto favourite_program_container {% if favourite_program %}text-yellow-400{% else %}text-white{% endif %} text-2xl{% if not program.public %} hidden{% endif %}">
+                <i class="fa fa-star p-4 bg-blue-300 rounded-full cursor-pointer hover:bg-blue-600"
+                    id="favourite_program_container_{{ program.id }}"
+                    data-starred="{{favourite_program}}"
```
Comment (hasan-sh) [inline templates/htmx-program.html:34]: This should fix it!

```suggestion
            <div
                id="favourite_program_container_{{ program.id }}"
                data-starred="{{favourite_program}}"
                class="ml-auto favourite_program_container {% if favourite_program %}text-yellow-400{% else %}text-white{% endif %} text-2xl{% if not program.public %} hidden{% endif %}">
                <i class="fa fa-star p-4 bg-blue-300 rounded-full cursor-pointer hover:bg-blue-600"
```

**Review by hasan-sh — CHANGES_REQUESTED** [review]
Hi, i remember making it possible to update the UI from TS; https://github.com/hedyorg/hedy/commit/ffa813a5ee9788ea81607dceff828125e298ad58#diff-810bc5b30fefd1c7d50d825f1331efb7cda58a958fdbe93d015904aed1735a47R29-R35


Why did you move the data-starred attribute to the i element instead of the container? Now you the ui isn't updating as it was! 

I suggested a change that fixes the issue, but do take a look at the test and whether anything else depends on what i changed!

**Comment by Annelein** [comment]
> Hi, i remember making it possible to update the UI from TS; [ffa813a#diff-810bc5b30fefd1c7d50d825f1331efb7cda58a958fdbe93d015904aed1735a47R29-R35](https://github.com/hedyorg/hedy/commit/ffa813a5ee9788ea81607dceff828125e298ad58#diff-810bc5b30fefd1c7d50d825f1331efb7cda58a958fdbe93d015904aed1735a47R29-R35)
> 
> Why did you move the data-starred attribute to the i element instead of the container? Now you the ui isn't updating as it was!

I did this because the test was failing because it would miss click because this is the space the container takes up, so now the test fails again:
Screenshot 2024-04-08 at 09 31 38

**Comment by hasan-sh** [comment]
@Annelein the `set_favourite_program` uses need to be fixed. Try to make a program private after favouriting it.

**Review by hasan-sh — APPROVED** [review]

**Pull request merged**